internal/runtimehandlerhooks/high_performance_hooks_linux.go

```diff
@@ -204,17 +216,123 @@ func annotationValueDeprecationWarning(annotation string) string {
 	return fmt.Sprintf("The usage of the annotation %q with value %q will be deprecated under 1.21", annotation, "true")
 }
 
-// disableCPULoadBalancing relies on the cpuset cgroup to disable load balancing for containers.
+// setCPULoadBalancing relies on the cpuset cgroup to disable load balancing for containers.
 // The requisite condition to allow this is `cpuset.sched_load_balance` field must be set to 0 for all cgroups
 // that intersect with `cpuset.cpus` of the container that desires load balancing.
 // Since CRI-O is the owner of the container cgroup, it must set this value for
 // the container. Some other entity (kubelet, external service) must ensure this is the case for all
 // other cgroups that intersect (at minimum: all parent cgroups of this cgroup).
-func disableCPULoadBalancing(containerManagers []cgroups.Manager) error {
+func setCPULoadBalancing(c *oci.Container, containerManagers []cgroups.Manager, enable bool) error {
 	if node.CgroupIsV2() {
-		return fmt.Errorf("disabling CPU load balancing on cgroupv2 not yet supported")
+		return setCPULoadBalancingV2(c, containerManagers, enable)
+	}
+	if !enable {
+		return disableCPULoadBalancingV1(containerManagers)
+	}
+	// There is nothing to do in cgroupv1 to re-enable load balancing
+	return nil
+}
+
+// On cgroupv2 systems, a new kernel API has been added to support load balancing
+// in a "remote" partition, layers away from the root cgroup.
+// This is done with a special file `cpuset.cpus.exclusive` which can be written to
+// request that cpuset be on standby for use by a cgroup that desires to be a partition.
+// To do this, each parent of the final cgroup must also have this value in the cpuset.cpus.exclusive,
+// and the final cgroup must have cpuset.cpus.partition = isolated.
+// This will cause the kernel to put that cpuset in a separate scheduling domain.
+// While this requires CRI-O to write to cgroups it does not own, it would be cumbersome to teach
+// other components in the system (kubelet/cpumanager) which cpu is newly set to exclusive each time
+// a pod request load balancing disabled.
+// Thus, this implementation assumes a certain amount of ownership CRI-O takes over this field. This
+// ownership may not be real in the future.
+// Another note on cgroup ownership: currently, CRI-O overwrites cpuset.cpus, which is a field managed by systemd.
+// It's possible systemd will clobber the value CRI-O sets.
```

Comment:
Should verify this behavior and it won't apply to crun, right?

Reply:
yeah I personally haven't seen it but we did hit this with cpu quota so it's possible in the future. also note: if systemd does eventually do this, crun would be effected because the parent cgroups are still created by systemd and require this dual ownership. 

I have adapted the PR to account for this though